### PR TITLE
8913 add sample resolution for step 2 of voluntary dissolution

### PR DESCRIPTION
**Pull request by argush3**
*Issue #:* /bcgov/entity#8913

*Description of changes:*

* Added step 2(special resolution/complete resolution) for Coops and Corps.  Note that this does not include the file uploader for Coops which will be done in another ticket.
* Added review and confirm section for step 2

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/resources/Dissolutions/communityContributionCompany.ts [hunk 1]
```diff
@@ -1,14 +1,114 @@
...
+        id: 'step-2-btn',
+        step: 2,
+        icon: 'mdi-handshake',
+        text: 'Complete \nResolution',
+        to: RouteNames.CREATE_RESOLUTION,
+        component: ViewComponentNames.CREATE_RESOLUTION
+      }
+    }
+    return item
+  }),
```
[the remark this answers is not in the document]
Comment (argush3) [reply src/resources/Dissolutions/communityContributionCompany.ts]: rebased so this should be gone now.

File: src/components/CreateResolution/UploadResolution.vue [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+#sample-resolution-section {
+  a {
+    text-decoration: none;
+  }
+}
+
+::v-deep #confirm-resolution-section .v-input__slot {
+  align-items: unset !important;
+  //height: unset !important;
```
[the remark this answers is not in the document]
Comment (argush3) [reply src/components/CreateResolution/UploadResolution.vue]: removed

File: src/views/Dissolutions/ReviewConfirmDissolution.vue [hunk 3]
```diff
@@ -239,7 +267,8 @@ export default class ReviewConfirmDissolution extends Mixins(DateMixin) {
 }
 
 .section-container {
-  padding: 1.5rem 2rem;
+  //padding: 1.5rem 2rem;
```
[the remark this answers is not in the document]
Comment (argush3) [reply src/views/Dissolutions/ReviewConfirmDissolution.vue]: removed

File: src/store/actions/actions-model.ts [hunk 4]
```diff
@@ -170,6 +170,14 @@ export const setMemorandumStepValidity: ActionIF = ({ commit }, validationDetail
   commit('mutateMemorandumStepValidity', validationDetail)
 }
 
+export const setResolution = ({ commit }, resolution: CreateResolutionIF) => {
+  commit('mutateResolution', resolution)
+}
+
+export const setResolutionStepValidity: ActionIF = ({ commit }, validationDetail: ValidationDetailIF): void => {
```
[the remark this answers is not in the document]
Comment (argush3) [reply src/store/actions/actions-model.ts:185]: will update to `setResolutionStepValidationDetail`

File: src/resources/Dissolutions/benefitCompany.ts [hunk 5]
```diff
@@ -1,14 +1,114 @@
...
+      ]
+    },
+    helpSection: {
+      header: `Help with Resolution`,
+      helpText: {
+        section1: {
+          items: [
+            {
+              type: ItemTypes.TEXT,
+              value: 'A Company can pass a director\'s resolution if:'
```
[the remark this answers is not in the document]
Comment (argush3) [reply src/resources/Dissolutions/benefitCompany.ts:69]: Added comment to ticket regarding all the proper noun instances and tagged Tracey.  Will see what she comes back with.

File: src/store/actions/actions-model.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (argush3) [reply src/store/actions/actions-model.ts]: updated relevant function names to indicate validation detail is being updated/set

File: src/components/CreateResolution/UploadResolution.vue [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+        {{getCreateResolutionResource.sampleFormSection.text}}
+      </p>
+      <v-card flat id="sample-resolution-card" class="pt-7 pb-5 pl-6">
+        <v-row>
+          <v-col id="sample-resolution-card-left-col" cols="1" class="pt-6" >
+          </v-col>
+          <v-col id="sample-resolution-card-center-col" cols="3" class="pl-8 pt-6">
+            <div>
+              <!-- Note: only able to resolve to preview image in assets folder properly when implemented this way. -->
+              <img :src="require(`@/assets/images/${getCreateResolutionResource.sampleFormSection.previewImagePath}`)"
```
[the remark this answers is not in the document]
Comment (argush3) [reply src/components/CreateResolution/UploadResolution.vue]: updated to use a getter.  was able to use `require` but need to use `require.context` first to get a context for the image assets folder to build the final image path.

File: src/views/Dissolutions/ReviewConfirmDissolution.vue [hunk 7]
```diff
@@ -124,14 +124,33 @@
...
+            <v-row no-gutters>
+              <v-col md="1">
+                <v-icon class="success-chk">mdi-check</v-icon>
+              </v-col>
+              <v-col md="11" id="file-name-col">
+                <span>FILE_NAME_PLACEHOLDER</span>
+              </v-col>
+            </v-row>
+          </div>
+        </section>
```
[the remark this answers is not in the document]
Comment (argush3) [reply src/views/Dissolutions/ReviewConfirmDissolution.vue]: cleaned this up a bit.

File: src/resources/Dissolutions/benefitCompany.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (argush3) [reply src/resources/Dissolutions/benefitCompany.ts:69]: Tracey got back to me.  This was her response:

>hi @argush3 - yes, capitalization rules are often not 100% consistent, but we decided to not capitalize these items for this app. so they are ok as is.

File: src/resources/Dissolutions/communityContributionCompany.ts [hunk 8]
```diff
@@ -1,14 +1,114 @@
...
     filingTypeCode: FilingCodes.DISSOLUTION_VOLUNTARY
+  },
+  createResolution: {
+    reviewConfirmHeader: 'Resolution',
+    introSection: {
+      header: '1. Resolution',
+      items: [
+        {
+          type: ItemTypes.TEXT,
+          value: 'Before submitting your voluntary dissolution you must <b>complete, sign, and date</b> a '
```
[the remark this answers is not in the document]
Comment (argush3) [reply src/resources/Dissolutions/communityContributionCompany.ts:47]: as per reply regarding first instance of this.

File: src/components/CreateResolution/UploadResolution.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply src/components/CreateResolution/UploadResolution.vue]: I like it. It keeps the "complicated stuff" out of the template and contains it in a method 👍

File: src/resources/Dissolutions/benefitCompany.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply src/resources/Dissolutions/benefitCompany.ts:69]: OK.

File: src/components/CreateResolution/UploadResolution.vue [hunk 9]
```diff
@@ -0,0 +1,374 @@
...
+  }
+
+  private previewImageSource ():string {
+    // Note: the image file path did not resolve correctly when using the require function directly.  In order
+    // to get the image path resolving correctly, needed to get the image context first and use that to build
+    // the final image image file path
+    const images = require.context('@/assets/images/', false, /\.png$/)
+    const previewImageSrc = images('./' + this.getCreateResolutionResource.sampleFormSection.previewImagePath)
+    return previewImageSrc
+  }
```
Comment (cameron-freshworks) [inline src/components/CreateResolution/UploadResolution.vue:195]: Pretty cool!

Could this be simplified to `return images('./' + this.getCreateResolutionResource.sampleFormSection.previewImagePath)` ?

Comment (argush3) [reply src/components/CreateResolution/UploadResolution.vue]: i kind of do this on purpose as i like placing breakpoints on the following line.  i.e. `return previewImageSrc`  But I do think it's definitely not as elegant too.  Anyways, I updated it 😋

**Review by severinbeauvais — APPROVED** [review]

File: src/components/CreateResolution/UploadResolution.vue (hunk 9 above)
Comment (severinbeauvais) [reply src/components/CreateResolution/UploadResolution.vue]: ^^ same for me

Comment (cameron-freshworks) [reply src/components/CreateResolution/UploadResolution.vue]: 😁 All good either way, thanks!

**Review by cameron-freshworks — APPROVED** [review]
🌟

File: src/components/CreateResolution/UploadResolution.vue [hunk 10]
```diff
@@ -0,0 +1,370 @@
...
+            >
+              <div slot='label' v-html="confirmLabel"></div>
+            </v-checkbox>
+            <ul>
+              <li class="mt-1"
+                v-for="(item, index) in getCreateResolutionResource.confirmSection.items"
+                :key="index">
+                <v-row v-if="item.type === ItemTypes.PARTIAL_ITEMS">
+                  <v-col cols="1" class="pt-1"><v-icon>mdi-circle-small</v-icon></v-col>
+                  <v-col cols="11" class="ml-n12 pt-1 pb-1">
```
[the remark this answers is not in the document]
Comment (argush3) [reply src/components/CreateResolution/UploadResolution.vue]: fiddled around with this a bunch.  ended up putting `no-gutters` on row, using `cols="auto"` for column 1 and adding using vuetify spacing helper class for the right margin.

**Comment by argush3** [comment]
Going to merge this into `main` while i write unit tests.  Also, there might be changes after discussion with Thor and Linda.  But I want to get some of this core code into `main`.

**Pull request merged**